tools/engine_tool/test/build_plan_test.dart

```diff
@@ -546,6 +546,102 @@ void main() {
     );
   });
 
+  test('can provide a single extra "--gn-args"', () {
+    final testEnv = TestEnvironment.withTestEngine();
+    addTearDown(testEnv.cleanup);
+
+    final testConfig = TestBuilderConfig();
+    testConfig.addBuild(
+      name: 'linux/host_debug',
+      dimension: TestDroneDimension.linux,
+    );
+
+    final parser = ArgParser();
+    final builds = BuildPlan.configureArgParser(
+      parser,
+      testEnv.environment,
+      configs: {
+        'linux_test_config': testConfig.buildConfig(
+          path: 'ci/builders/linux_test_config.json',
+        ),
+      },
+      help: false,
+    );
```

Comment:
This whole section is duplicated across tests, can we factor it out?

Reply:
As this PR is a net positive and unblocks the Dart SDK team, let me land as-is and I'll follow-up shortly in https://github.com/flutter/flutter/issues/157870.